File: leetcode/binary_tree/findTilt.js
var findTilt = function(root) {
  function postOrder(root) {
    if (root === null) return 0;

    const left = root.left;
    const right = root.right;
    const leftResult = postOrder(left);
    const rightResult = postOrder(right);

    sum += Math.abs(leftResult - rightResult);

    return root.val + leftResult + rightResult;
  }

  let sum = 0;

  postOrder(root);

  return sum;
};

File: leetcode/binary_tree/find_tilt.rb
def find_tilt_helper(root, array)
  return 0 if root.nil?

  left = root.left
  right = root.right
  left_result = find_tilt_helper(left, array)
  right_result = find_tilt_helper(right, array)
  val = root.val

  array[0] += (left_result - right_result).abs

  return val + left_result + right_result
end

def find_tilt(root)
  return 0 if root.nil?

  array = [0]
  find_tilt_helper(root, array)

  array[0]
end
